adal/tests/Test.ADAL.NET.UIAutomation/CoreMobileMSALTests.cs

```diff
@@ -0,0 +1,134 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Test.Microsoft.Identity.Core.UIAutomation.infrastructure;
+
+namespace Test.MSAL.NET.UIAutomation
+{
+    /// <summary>
+    /// Contains the core test functionality that will be used by Android and iOS tests
+    /// </summary>
+    public static class CoreMobileMSALTests
+    {
+
+        private const string MSGraph = "https://graph.microsoft.com";
+        private const string UIAutomationAppV2 = "1e245a30-49aa-43eb-b9c1-c11b072cc92b";
+
+        /// <summary>
+        /// Runs through the standard acquire token flow
+        /// </summary>
+        /// <param name="controller">The test framework that will execute the test interaction</param>
+        public static void AcquireTokenTest(ITestController controller)
+        {
+            var user = prepareForAuthentication(controller);
+
+            SetInputData(controller, UIAutomationAppV2, "User.Read");
+
+            PerformSignInFlowFlow(controller, user);
+
+            //Verify result. Test results are put into a label
+            Assert.IsTrue(controller.GetText("testResult").Contains("Result: Success"));
+        }
+
+        /// <summary>
+        /// Runs through the standard acquire token flow
+        /// </summary>
+        /// <param name="controller">The test framework that will execute the test interaction</param>
+        public static void AcquireTokenSilentTest(ITestController controller)
+        {
+            var user = prepareForAuthentication(controller);
+
+            SetInputData(controller, UIAutomationAppV2, "User.Read");
+
+            PerformSignInFlowFlow(controller, user);
+
+            Assert.IsTrue(controller.GetText("testResult").Contains("Result: Success"));
+
+            SetInputData(controller, UIAutomationAppV2, "User.Write");
+
+            controller.Tap("acquireTokenSilent");
+
+            //Verify result. Test results are put into a label
+            Assert.IsTrue(controller.GetText("testResult").Contains("Result: Success"));
+        }
+
+        /// <summary>
+        /// Runs through the standard acquire token interactive flow
+        /// </summary>
+        /// <param name="controller">The test framework that will execute the test interaction</param>
+        public static void AcquireTokenADFSvXInteractiveMSALTest(ITestController controller, FederationProvider federationProvider, bool isFederated)
+        {
+            var user = prepareForAuthentication(controller);
+
+            SetInputData(controller, UIAutomationAppV2, "User.Read");
+
+            PerformSignInFlowFlow(controller, user);
+
+            //Verify result. Test results are put into a label
+            Assert.IsTrue(controller.GetText("testResult") == "Result: Success");
+        }
+
+        private static IUser prepareForAuthentication(ITestController controller)
+        {
+            //Clear Cache
+            controller.Tap("Cache");
```

Comment:
>"Cache" [](start = 27, length = 7)

see related comments to const for id's on various controls on the form/page (several similar to this one). I suggest you create a new PR with that or ensure that is part of the unified test app work.